### ISSUE TITLE
Wait.go daesonset and version limitations?

### ISSUE THREAD

**Issue by paulgmiller** [helm/helm#5219]
Was look at using Helm upgrade/install --wait for our continuous deployment but I had two concerns.

1) should waiting daemonset ignore nodes that are not ready and/or honor max unavailable. The concern is that in large node pools you might never succeed as at most points in time you at least have 1 unready node.

2) In deployments, daesmonsets and stateful sets should some sort of label be used to select only the current revisionnumber  when grabbing replicaset/pods)? It seems like without it we could pick up older or newer replicasets/pods and wait on the wrong thing. That _may _ be what is happening in [this issue](https://github.com/helm/helm/issues/4280). Maybe this is already is handled and I'm missing it.

Happy to contribute some code but wanted to see if you thought this was valid and that wait was worth investing in (maybe there's a diferent solution preferred for monitoring releases) before I went forward.

**Comment by paulgmiller** [comment]
Also I guess this could be lumped in with #3176

**Comment by bacongobbler** [comment]
cc @thomastaylor312

**Comment by paulgmiller** [comment]
Okay let me spin up a PR for the first. Would probably start with max unavailable since it's easier than querying node status. 

You're definitely using [deploymentutil.GetNewReplicaSet](https://github.com/kubernetes/kubernetes/blob/2cbb16bc8dd456c5db72c1667926abdbc87c32c7/pkg/controller/deployment/util/deployment_util.go) and that seems use the spec hash  to match `EqualIgnoreHash(&rsList[i].Spec.Template, &deployment.Spec.Template) `
So that's good since you hold onto the deployment spec. Maybe we could do the same with the pods from rs/stateful set but I'll leave that for a seperate pr.

**Comment by paulgmiller** [comment]
Sorry for dissapearing .I learned alot about daemonsets status in kubernetes sicne that including the fact that they expect daemonsets to run on notready nodes which makes things a bit harder.  Would need

Still interested in cleaning up the wait.go code a bit and making it more like kubectl rollout status as described in 3176. 

Even setting notready nodes aside there's definitely an issue where wait can finish before any pods are terminated in a rolling daemonset. 

```
019-03-25T21:21:46.2610801Z 2019-03-25 21:21:46.253 - DEBUG    - ==> v1/Pod(related)

2019-03-25T21:21:46.2611856Z 2019-03-25 21:21:46.254 - DEBUG    - NAME READY STATUS RESTARTS AGE

2019-03-25T21:21:46.2613012Z 2019-03-25 21:21:46.254 - DEBUG    - log-tailer-4565j 3/3 Terminating 15 24d
```

**Comment by bacongobbler** [comment]
can this be closed with #5971?

cc @thomastaylor312

**Comment by thomastaylor312** [comment]
@bacongobbler yep, with the latest daemonset status objects and the updates from #5971, this can be closed!
